test/lib/y2users/equatable_test.rb

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env rspec
+
+# Copyright (c) [2021] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require_relative "test_helper"
+require "y2users/equatable"
+
+describe Y2Users::Equatable do
```

Comment:
In general LGTM.

np: as it implements the **hash** method I miss a unit test covering it and also for the **#==** one which could be a shared example for both **eql?** and **#==**.

Reply:
The `#hash` method in indirect tested through `#eql?`. Its test would be basically the same. And I am not sure for `#==`, they are alias. But ok, I can run the very same tests for it.

Reply:
Yep, both are indirect tested, it would be more about documentation than other thing but it looks good to me as it is.